duckduckgo/iOS
# iOS InContext Email Protection Signup

Task/Issue URL: https://app.asana.com/0/72649045549333/1202940260106121/f
Tech Design URL:
CC:

**Description**:
Adds Email InContext Signup protection to iOS



**Steps to test this PR**:
1. Starting with a clean install, visit `fill.dev/form/registration-email`
2. Confirm the grey dax icon is visible in the email field
3. Tap the text field (not the icon) and confirm the keyboard is displayed
4. Tap the icon and confirm the prompt to sign up for email protection is presented
5. Tap 'x' to dismiss the prompt and confirm the dax icon is still visible & grey
6. Tap the icon again, and this time select "Don't show again"
7. Confirm the dax icon is no longer visible and that refreshing the page does not bring it back
8. In the app go to Settings > Debug and tap the row "Reset Email Protection InContext Signup"
9. Return to `fill.dev/form/registration-email` and refresh the page. The grey dax icon should now be back
10. Tap the icon again, and this time select "Protect my email"
11. Choose "Get Started" to follow the flow to create a new duck address
12. Confirm that at the end of the sign up process, when dismissing the modal either from the 'Done' button or the 'Try it now', the prompt to chose a duck address is presented and that selecting an address populates the web form
13. Sign out and test the sign in flow (including tapping the link from email to launch the app)



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
